apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileTaskExecutionService.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.profile;
+
+import org.apache.skywalking.apm.agent.core.boot.BootService;
+import org.apache.skywalking.apm.agent.core.boot.DefaultNamedThreadFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Profile task executor, use {@link #addProfileTask(ProfileTask)} to add a new profile task.
+ *
+ * @author MrPro
+ */
+public class ProfileTaskExecutionService implements BootService {
+
+    private static volatile ScheduledExecutorService PROFILE_TASK_READY_SCHEDULE = Executors.newScheduledThreadPool(15, new DefaultNamedThreadFactory("PROFILE-TASK-READY-SCHEDULE"));
```

Comment:
Hi, I suggest will `ProfileTaskExecutionService` disabled by default, enabled by the user through the displayed configuration.

Reply:
Sorry, I'm not sure what you mean? Why need to disable it? I need to use it to process the profile task. Do you mean to make a switch, user decides sniffer can enable profile?

Reply:
> Do you mean to make a switch, user decides sniffer can enable profile?

Yes.

Reply:
@arugal Why make it disabled in default? Could you explain a little?

Reply:
please check.
I add a new agent switch config `SW_AGENT_ACTIVE_PROFILE`.
I will be working on the query profile tasks. If disable the query, the sniffer will not have any profile task need to be executed.

Reply:
please check. I have added a new agent switch config `agent.active_profile`.
It will be working on the query profile tasks. If disable the query, the sniffer will not have any profile task need to be executed.

Reply:
Agent config? So if the user wants to enable profiling, he must restart the agent? Unacceptable to me